File: src/main/java/cz/vhromada/catalog/web/validator/TimeValidator.java
package cz.vhromada.catalog.web.validator;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

import cz.vhromada.catalog.web.fo.TimeFO;
import cz.vhromada.catalog.web.validator.constraints.Time;

import org.springframework.util.StringUtils;

/**
 * A class represents show validator for time constraint.
 *
 * @author Vladimir Hromada
 */
public class TimeValidator implements ConstraintValidator<Time, TimeFO> {

    @Override
    public void initialize(final Time time) {
    }

    @Override
    public boolean isValid(final TimeFO time, final ConstraintValidatorContext constraintValidatorContext) {
        if (time == null) {
            return false;
        }
        if (StringUtils.isEmpty(time.getHours()) || StringUtils.isEmpty(time.getMinutes()) || StringUtils.isEmpty(time.getSeconds())) {
            return true;
        }

        final int length = Integer.parseInt(time.getHours()) + Integer.parseInt(time.getMinutes()) + Integer.parseInt(time.getSeconds());
        return length > 0;
    }

}

File: src/main/kotlin/cz/vhromada/catalog/web/validator/TimeValidator.kt
package cz.vhromada.catalog.web.validator

import cz.vhromada.catalog.web.fo.TimeFO
import cz.vhromada.catalog.web.validator.constraints.Time
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

/**
 * A class represents show validator for time constraint.
 *
 * @author Vladimir Hromada
 */
class TimeValidator : ConstraintValidator<Time, TimeFO> {

    override fun isValid(time: TimeFO?, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        if (time == null) {
            return false
        }
        if (time.hours.isNullOrBlank() || time.minutes.isNullOrBlank() || time.seconds.isNullOrBlank()) {
            return true
        }

        val length = time.hours!!.toInt() + time.minutes!!.toInt() + time.seconds!!.toInt()
        return length > 0
    }

}
